mono/metadata/boehm-gc.c

```diff
@@ -354,6 +354,18 @@ mono_gc_thread_attach (MonoThreadInfo* info)
 	return info;
 }
 
+void
+mono_gc_thread_detach (MonoThreadInfo *p)
+{
+	/* Detach without threads lock as Boehm
+	 * will take it's own lock internally. Note in
+	 * on_gc_notification we take threads lock after
+	 * Boehm already has it's own lock. For consistency
+	 * always take lock ordering of Boehm then threads.
+	 */
+	GC_unregister_my_thread ();
```

Comment:
I'm not sure I understand why we don't have to do the mono_handle_stack_free stuff we aredoing in  boehm's mono_gc_thread_detach_with_lock. Is there something special about the threads that will be calling this function?

Reply:
This PR makes `thread_detach` call `mono_gc_thread_detach`

`thread_detach` is a callback that is called from `unregister_thread`, and `thread_detach_with_lock` is called immediately after:

https://github.com/mono/mono/blob/439091eddf514ed259733fed41d554655d889bae/mono/utils/mono-threads.c#L559-L571

Reply:
I see. The names just made me think they are the same thing with/with out locks, but its more like the *part* of thread detaching we can do with out a lock, then the part we can do with a lock.